### PR TITLE
Jwt controller action

### PR DESCRIPTION
**Pull request by winescout**
NOTICE
==========
This is a follow on PR for the Devise upgrade PR.  Have a look [here](https://github.com/notch8/Hours/pull/24/files) to get a lot of the boiler plate in the PR merged first.

The list of files without the Devise stuff:
===============================
      -  modified:   .env
      -  modified:   Gemfile
      -  modified:   Gemfile.lock
      -  modified:   app/controllers/application_controller.rb
      -  new file:   app/controllers/sessions_controller.rb
      -  new file:   app/models/jwt_blacklist.rb
      -  modified:   app/models/user.rb
      -  modified:   config/application.rb
      -  modified:   config/initializers/devise.rb
      -  modified:   config/routes/subdomain_present.rb
      -  new file:   db/migrate/20190831203215_create_jwt_blacklist.rb
      -  modified:   db/schema.rb
      -  new file:   spec/factories/jwt_blacklists.rb
      -  new file:   spec/features/user_sign_in_with_jwt_spec.rb

### CODE REVIEW

**Comment by orangewolf** [comment]
I think the jwt key default in .env.development is totally appropriate. I think the other changes (discussed in the devise PR) need to be pulled out please.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :reset_session
+  skip_before_action :verify_authenticity_token
```
Comment (orangewolf) [inline app/controllers/application_controller.rb]: this seems like a bug?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    end
+  end
+
+  private
+
+  def respond_with(resource, _opts = {})
+    respond_to do |format|
+      format.html{ super(resource, _opts) }
+      format.json do
+        render json: resource, headers: {'Authorization':'aaaaaaa'}
```
Comment (orangewolf) [inline app/controllers/sessions_controller.rb]: are we shouting in to the void?

File: config/routes/subdomain_present.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-devise_for :users, controllers: { registrations: "users/registrations" }
+devise_for :users, controllers: { registrations: "users/registrations", sessions: 'sessions' }
```
Comment (orangewolf) [inline config/routes/subdomain_present.rb]: this should be users/sessions (and yes that probably means the controller should move ;-)

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (winescout) [reply app/controllers/sessions_controller.rb]: yowza.  thanks.

File: config/routes/subdomain_present.rb (hunk 3 above)
Comment (winescout) [reply config/routes/subdomain_present.rb]: so,  there is no users/registrations controller at all.  In fact, there is no registration in this app at all

Comment (winescout) [reply config/routes/subdomain_present.rb]: That said.  I'll move it

Comment (orangewolf) [reply config/routes/subdomain_present.rb]: oh, that's odd. it does follow what we do in other apps so i didn't even question it

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (winescout) [reply app/controllers/application_controller.rb]: switched it over to null session

**Pull request merged**